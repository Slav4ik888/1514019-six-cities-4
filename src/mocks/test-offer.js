export const testOffer = {
  id: 1,
  isPremium: true,
  isFavourite: false,
  previewImage: `img/apartment-03.jpg`,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/room.jpg`,
    `img/studio-01.jpg`,
    `img/studio-photos.jpg`],
  amenities: [`Wifi`, `Heating`, `Kitchen`, `Cable TV`],
  bedrooms: 4,
  maxGuestsNumber: 5,
  description: `This cozy and complete apartment in the heart of New York is a typical, traditional and authentic feel for how New Yorkers live. I welcome all guests to stay at this place during their stay and visit of New York. 
  Please note that this apartment is on the sixth floor and there is no lift (elevator).
  The apartment is convenient for all guests, fitted with a kitchen, small dining area and two bedrooms – couples and groups of friends will find this accommodation most optimal.`,
  host: {
    photo: `img/avatar-angelina.jpg`,
    name: `Angelina Jonson`,
    super: true
  },
  price: 180,
  rating: 5,
  cardTitle: `Beautiful & luxurious apartment at great location`,
  offerType: `House`,
  coordinates: [48.856663, 2.351556],
  city: {
    zoom: 8,
    coordinates: [48.856663, 2.351556],
  },
  location: {
    zoom: 8,
  },
  reviews: [
    {
      id: 1,
      author: {
        photo: `img/avatar-angelina.jpg`,
        name: `Strange jober`
      },
      description: `small dining area and two bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095836000,
      rating: 3
    },
    {
      id: 2,
      author: {
        photo: ``,
        name: `Slava`
      },
      description: `dining area and five bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095936000,
      rating: 4.7
    },
  ],

};
