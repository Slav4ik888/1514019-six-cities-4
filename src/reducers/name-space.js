export const NameSpace = {
  DATA: `DATA`,
  TRAVEL: `TRAVEL`,
  USER: `USER`,
  FAVORITE: `FAVORITE`,
};
