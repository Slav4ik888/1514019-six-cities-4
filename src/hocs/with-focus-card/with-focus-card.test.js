import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import withFocusCard from './with-focus-card.js';
import {NameSpace} from '../../reducers/name-space.js';
import {testOffer} from '../../mocks/test-offer.js';

const mockStore = configureStore([]);

const MockComponent = () => <div/>;

const MockComponentWrapped = withFocusCard(MockComponent);

describe(`Render withFocusCard correctly`, () => {
  it(`Render withFocusCard correctly with focus`, () => {
    const store = mockStore({
      [NameSpace.TRAVEL]: {
        activeHoverOffer: null,
      }
    });
    const tree = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped
            onCardFocusEnter={() => (null)}
            onCardFocusLeave={() => {}}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render withFocusCard correctly without focus`, () => {
    const store = mockStore({
      [NameSpace.TRAVEL]: {
        activeHoverOffer: null,
      }
    });
    const tree = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped
            onCardFocusEnter={() => (testOffer)}
            onCardFocusLeave={() => {}}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

// npm run test.jest -- -u with-focus-card.test.js
// npm test with-focus-card.test.js
