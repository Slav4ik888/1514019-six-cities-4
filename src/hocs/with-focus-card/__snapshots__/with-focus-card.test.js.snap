// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFocusCard is rendered correctly 1`] = `<div />`;
